openvm/src/customize_exe.rs

```diff
@@ -145,71 +156,70 @@ pub fn customize(
         g: OpenVmField::<BabyBearField>::ZERO,
     };
 
-    let mut extensions = Vec::new();
-    let n_acc = config.autoprecompiles as usize;
-    let n_skip = config.skip_autoprecompiles as usize;
-
     tracing::info!("Adjust the program with the autoprecompiles");
 
-    // now the blocks have been sorted by cost
-    for acc_block in blocks.iter().skip(n_skip).take(n_acc) {
-        let CachedAutoPrecompile {
-            apc_opcode,
-            autoprecompile,
-            subs,
-        } = apc_cache.remove(&acc_block.start_idx).unwrap();
-
-        let new_instr = Instruction {
-            opcode: VmOpcode::from_usize(apc_opcode),
-            a: OpenVmField::<BabyBearField>::ZERO,
-            b: OpenVmField::<BabyBearField>::ZERO,
-            c: OpenVmField::<BabyBearField>::ZERO,
-            d: OpenVmField::<BabyBearField>::ZERO,
-            e: OpenVmField::<BabyBearField>::ZERO,
-            f: OpenVmField::<BabyBearField>::ZERO,
-            g: OpenVmField::<BabyBearField>::ZERO,
-        };
-
-        let pc = acc_block.start_idx;
-        let n_acc = acc_block.statements.len();
-        let (acc, new_instrs): (Vec<_>, Vec<_>) = program[pc..pc + n_acc]
-            .iter()
-            .enumerate()
-            .map(|(i, x)| {
-                let instr = x.as_ref().unwrap();
-                let instr = instr.0.clone();
-                if i == 0 {
-                    (instr, new_instr.clone())
-                } else {
-                    (instr, noop.clone())
-                }
-            })
-            .collect();
-
-        let new_instrs = new_instrs.into_iter().map(|x| Some((x, None)));
-
-        let len_before = program.len();
-        program.splice(pc..pc + n_acc, new_instrs);
-        assert_eq!(program.len(), len_before);
-
-        let is_valid_column = autoprecompile
-            .unique_references()
-            .find(|c| &*c.name == "is_valid")
-            .unwrap();
+    let extensions = blocks_with_apcs
```

Comment:
Diff looks big but I just turned a loop into a map. There is still a side effect on the program though, it gets modified to insert the new instructions.